### PR TITLE
Fix dragging when control has a rotated parent.

### PR DESCRIPTION
**Pull request by carolhmj**

### CODE REVIEW

File: guiEditor/src/diagram/workbench.tsx [hunk 1]
```diff
@@ -680,8 +681,24 @@ export class WorkbenchComponent extends React.Component<IWorkbenchComponentProps
             return true;
         }
 
-        guiControl.leftInPixels += newX;
-        guiControl.topInPixels += newY;
+        let totalRotation = 0;
+        let currentControl : Nullable<Control> = guiControl;
+        while (currentControl !== null && currentControl !== undefined) {
```
Comment (RaananW) [inline guiEditor/src/diagram/workbench.tsx:686]:
```suggestion
        while (currentControl) {
```

File: guiEditor/src/diagram/workbench.tsx [hunk 2]
```diff
@@ -666,7 +666,8 @@ export class WorkbenchComponent extends React.Component<IWorkbenchComponentProps
         } else if (this._constraintDirection === ConstraintDirection.Y) {
             newX = 0;
         }
-
+        let referenceAxis = new Vector2(newX, newY);
```
Comment (RaananW) [inline guiEditor/src/diagram/workbench.tsx:669]: nit - const

**Review by RaananW — COMMENTED** [review]
LGTM!

Just two small comments

File: guiEditor/src/diagram/workbench.tsx (hunk 2 above)
Comment (carolhmj) [reply guiEditor/src/diagram/workbench.tsx]: Updated!

**Review by RaananW — APPROVED** [review]

**Comment by RaananW** [comment]
waiting for @darraghjburke, in case he sees something

**Review by darraghjburke — APPROVED** [review]
looks great to me!

**Pull request merged**